nexus/db-queries/src/transaction_retry.rs

```diff
@@ -0,0 +1,141 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Helper types for performing automatic transaction retries
+
+use chrono::Utc;
+use oximeter::{types::Sample, Metric, MetricsError, Target};
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+// Identifies "which" transaction is retrying
+#[derive(Debug, Clone, Target)]
+struct Transaction {
+    name: String,
```

Comment:
I'm not sure "which CRDB server we're connected to" is stable, so I don't think it should be part of the Target.

I figured this would be enough to *start*, simply because seeing a transaction name on here means "it's worth making that transaction leaner" -- the timestamps should also paint a pretty relevant picture of timing.